pkg/logging/logging.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package fybriklogging
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+)
+
+// LogType indicates the types of log entries supported.  Logging packages typically provide separate function
+// calls for the different log types.
+type LogType string
+
+const (
+	ERROR LogType = "error" // Errors encountered in the code
+	INFO  LogType = "info"  // Info logs are things you want to tell the user or administrator which are not errors.
+)
+
+// VerbosityLevel allows one to decide how "chatty" you want your logs to be.
+type VerbosityLevel int
+
+const (
+	MANDATORY VerbosityLevel = 0
+	DEBUG     VerbosityLevel = 1
+	TRACE     VerbosityLevel = 2
+)
+
+// LogEntry defines the information for a single entry stored in the system logs.
+// It is assumed that the timestamp is added automatically and thus not included in the log entry structure.
+// Verbosity and log type are added by the log library via which the entries are sent to stdout or stderr.
+type LogEntry struct {
+	// Caller indicates the cluster, component and function that generated the message in the format "cluster/component/function"
```

Comment:
I think that logging the function is not necessary. This makes the filtering by component more difficult, depends on whether the logging was done by the inner function or the outer function receiving the response, might break after code refactoring and makes the caller string very long. It does not add extra value for developers either.

Reply:
@shlomitk1 Don't you think it has value for tracing purposes?

Reply:
I don't. First, it is not a full stack, as printed for segmentation fault. Second, for tracing you'd usually need to go to other logs (of the connectors, for example) to trace the problem.

Reply:
Removed